gtfu/src/main/java/gtfu/tools/TripListGenerator.java

```diff
@@ -329,33 +349,28 @@ public RouteData(String agencyID, String routeShortName, String routeLongName) {
 }
 
 class TripListEntry implements Comparable<TripListEntry> {
-    String headSign;
     String serviceID;
     String routeID;
-    String routeShortName;
-    String routeLongName;
     String firstStopID;
     String tripID;
-    String shapeID;
     String name;
     int departureTime;
 
-    public TripListEntry(TripData data, RouteData rdata, String firstStopID, String tripID, int departureTime, String nameField) {
-        headSign = data.headSign;
+    public TripListEntry(TripData data, RouteData rdata, String firstStopID, String tripID, int departureTime, String nameField, String direction, boolean useDirection) {
         serviceID = data.serviceID;
         routeID = data.routeID;
-        routeShortName = rdata.routeShortName;
-        routeLongName = rdata.routeLongName;
-        shapeID = data.shapeID;
 
         if (nameField.equals("headsign")){
-            name = headSign;
+            name = data.headSign;
         } else if (nameField.equals("route_short_name")) {
-            name = routeShortName;
+            name = rdata.routeShortName;
         } else if (nameField.equals("route_long_name")) {
-            name = routeLongName;
+            name = rdata.routeLongName;
         } else if (nameField.equals("shape_id")) {
-            name = shapeID;
+            name = data.shapeID;
+        }
+        if (useDirection){
+            name += " - " + direction;
         }
         if (name == null || name == "") {
```

Comment:
As discussed, using the == operator on strings is a bad idea. Also, there is Util.isEmpty() which I think does both of the above checks. Please fix and also check for other occurrences of == with strings.

Reply:
shame!